src/Cmd/Add.cc

```diff
@@ -0,0 +1,235 @@
+  std::vector<std::string> newDeps = {};
+
+  bool isSystemDependency = false;
+  std::string version; // Only used with system-dependencies
+
+  std::string tag;
+  std::string rev;
+  std::string branch;
+
+  for (auto itr = args.begin(); itr != args.end(); ++itr) {
+    if (const auto res = Cli::handleGlobalOpts(itr, args.end(), "add")) {
+      if (res.value() == Cli::CONTINUE) {
+        continue;
+      } else {
+        return res.value();
+      }
+    } else if (*itr == "--sys") {
+      isSystemDependency = true;
+    } else if (*itr == "--version" || *itr == "-v") {
+      if (auto nextArg = getNextArg(itr, args.end())) {
+        version = *nextArg;
+      } else {
+        return ADD_CMD.missingArgumentForOpt(*--itr);
+      }
+    } else if (*itr == "--tag") {
+      if (auto nextArg = getNextArg(itr, args.end())) {
+        tag = *nextArg;
+      } else {
+        return ADD_CMD.missingArgumentForOpt(*--itr);
+      }
+    } else if (*itr == "--rev") {
+      if (auto nextArg = getNextArg(itr, args.end())) {
+        rev = *nextArg;
+      } else {
+        return ADD_CMD.missingArgumentForOpt(*--itr);
+      }
+    } else if (*itr == "--branch") {
+      if (auto nextArg = getNextArg(itr, args.end())) {
+        branch = *nextArg;
+      } else {
+        return ADD_CMD.missingArgumentForOpt(*--itr);
+      }
+    } else {
+      handleDependency(newDeps, std::string(*itr));
+    }
+  }
+
+  toml::value depData = toml::table{};
+
+  if (isSystemDependency) {
+    if (version.empty()) {
+      logger::error("The `--version` option is required for system dependencies"
+      );
+      return EXIT_FAILURE;
+    }
+    depData["version"] = version;
+    depData["system"] = true;
+  } else {
+    if (!tag.empty()) {
+      depData["tag"] = tag;
+    }
+    if (!rev.empty()) {
+      depData["rev"] = rev;
+    }
+    if (!branch.empty()) {
+      depData["branch"] = branch;
+    }
+  }
+
+  auto data = toml::parse(getManifestPath());
+  auto& deps = toml::find<toml::table>(data, "dependencies");
+
+  for (const auto& dep : newDeps) {
+
+    if (!isSystemDependency) {
+      const std::string gitUrl = getDependencyGitUrl(dep);
+      const std::string depName = getDependencyName(dep);
+      if (gitUrl.empty() || depName.empty()) {
+        return EXIT_FAILURE;
+      }
+
+      deps[depName] = depData;
+      deps[depName]["git"] = gitUrl;
+    } else {
+      deps[dep] = depData;
+    }
+  }
+
+  std::ofstream ofs(getManifestPath());
+  ofs << toml::format(data);
+
+  logger::info("Added", "to the poac.toml");
+
+  return 0;
+}
+
+std::string
+getDependencyGitUrl(const std::string& dep) {
+  if (dep.find("://") == std::string::npos) {
+    // check if atleast in "user/repo" format
+    if (dep.find('/') == std::string::npos) {
+      logger::error("Invalid dependency: " + dep);
+      return "";
+    }
+
+    return "https://github.com/" + dep + ".git";
+  }
+  return dep;
+}
+
+std::string
+getDependencyName(const std::string& dep) {
```

Comment:
I think we can use const StringRef for dep. Also, for the return type.

Reply:
Resolved for dep. The return type is used as the key in toml_table which doesn't support StringRef